Description: Allow compiler flags to be passed through to build system
Author: Afif Elghraoui <afif@ghraoui.name>
Forwarded: no
Last-Update: 2015-11-03
--- bcftools.orig/Makefile
+++ bcftools/Makefile
@@ -32,8 +32,8 @@
 BGZIP  = /usr/bin/bgzip
 TABIX  = /usr/bin/tabix
 
-CC       = gcc
-CFLAGS   = -g -Wall -Wc++-compat -O2
+CC       ?= gcc
+CFLAGS   += -g -Wall -Wc++-compat -O2
 DFLAGS   =
 OBJS     = main.o vcfindex.o tabix.o \
            vcfstats.o vcfisec.o vcfmerge.o vcfquery.o vcffilter.o filter.o vcfsom.o \
@@ -52,6 +52,8 @@
     LDLIBS  = -lgsl -lgslcblas
 endif
 
+LDLIBS += $(LDFLAGS)
+
 prefix      = /usr/local
 exec_prefix = $(prefix)
 bindir      = $(exec_prefix)/bin
@@ -100,7 +102,7 @@
 PLUGINM = $(PLUGINC:.c=.mk)
 
 %.so: %.c version.h version.c
-	$(CC) $(CFLAGS) $(INCLUDES) -fPIC -shared -o $@ version.c $< -lhts
+	$(CC) $(CFLAGS) $(INCLUDES) -fPIC -shared -o $@ version.c $<
 
 -include $(PLUGINM)
 
